mypy/types.py

```diff
@@ -117,6 +117,15 @@
     'typing_extensions.Annotated',
 )
 
+# We use this constant in various places when checking `tuple` subtyping:
```

Comment:
Not related to your diff, but I wonder why Sized and Collection aren't in here.

Reply:
Do we need to open a new issue? 🤔

Reply:
Yes probably. I guess what this affects is whether a heterogeneous tuple type matches `Sized` or `Collection`?